src/EditorFeatures/Core/AddImports/AbstractAddImportsPasteCommandHandler.cs

```diff
@@ -35,11 +39,16 @@ internal abstract class AbstractAddImportsPasteCommandHandler : IChainedCommandH
 
         private readonly IThreadingContext _threadingContext;
         private readonly IGlobalOptionService _globalOptions;
+        private readonly IAsynchronousOperationListener _listener;
 
-        public AbstractAddImportsPasteCommandHandler(IThreadingContext threadingContext, IGlobalOptionService globalOptions)
+        public AbstractAddImportsPasteCommandHandler(
+            IThreadingContext threadingContext,
+            IGlobalOptionService globalOptions,
+            IAsynchronousOperationListenerProvider listenerProvider)
         {
             _threadingContext = threadingContext;
             _globalOptions = globalOptions;
+            _listener = listenerProvider.GetListener(FeatureAttribute.GoToDefinition);
```

Comment:
not the right listener :)   there should be a listener for AddImportsOnPaste (and, if not, you can add one).

Reply:
we shoudl also def have an integration test added to validate this works

Reply:
woops :) copy-paste failure